### PR TITLE
chore(ci): tests fix + ci configuration

### PR DESCRIPTION
**Pull request by goooseman**
This PR

1. Changes GitHub Actions workflow to run CI on every Pull Request
2. Resolves a problem with "Throttle CPU" test

### CODE REVIEW

**Comment by Rowno** [comment]
I don't think the test is fixed yet 😅

**Comment by goooseman** [comment]
@Rowno as my Papa was always telling me: do not do any changes in a rush 😂

Should be fixed right now 🤞

**Comment by goooseman** [comment]
@Rowno PR is ready to be merged
I will update another one as soon as this one is merged

File: test/cli.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ test('throttles CPU', async (t) => {
 
   // difference should be more then 2 times
```
Comment (Rowno) [inline test/cli.js]: Does this comment still make sense?

Comment (goooseman) [reply test/cli.js]: Fixed

**Pull request merged**